### PR TITLE
ansible collection repo sync from hub

### PR DESCRIPTION
**Pull request by pondrejk**
```
pytest tests/foreman/cli/test_repository.py -k test_positive_sync_ansible_collecion_from_gallaxy
=================================================== test session starts ====================================================
platform linux -- Python 3.9.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/pondrejk/Documents/robottelo, configfile: pyproject.toml
plugins: forked-1.3.0, services-2.2.1, reportportal-5.0.8, mock-3.6.1, cov-2.12.1, xdist-2.3.0, ibutsu-1.16
collected 238 items / 236 deselected / 2 selected                                                                          

tests/foreman/cli/test_repository.py ..                                                                              [100%]

================================= 2 passed, 236 deselected, 6 warnings in 80.49s (0:01:20) =================================
```

Ready for review, though putting wip to wait for the associated satelliteqe-jenkins PR

### CODE REVIEW

File: conf/ansible_hub.yaml.template [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ANSIBLE_HUB:
```
Comment (mshriver) [inline conf/ansible_hub.yaml.template:1]: Please define a validator for this namespace:

https://github.com/SatelliteQE/robottelo/blob/master/robottelo/config/validators.py

File: tests/foreman/cli/test_repository.py [hunk 2]
```diff
@@ -2385,17 +2385,24 @@ class TestAnsibleCollectionRepository:
...
+            {
+                'content-type': 'ansible_collection',
+                'url': settings.ansible_hub.url,
+                'ansible-collection-auth-token': settings.ansible_hub.token,
+                'ansible-collection-auth-url': settings.ansible_hub.sso_url,
+                'ansible-collection-requirements': '{collections: \
+                                                         [redhat.satellite_operations ]}',
+            },
+        ],
+        ids=['ansible_gallaxy', 'ansible_hub'],
```
Comment (mshriver) [inline tests/foreman/cli/test_repository.py:2405]: :+1: Great call setting ids here instead of just using `**parametrized` to give them integer indices.

Comment (mshriver) [reply tests/foreman/cli/test_repository.py:2405]: One `l` in `ansible_galaxy` though

File: tests/foreman/cli/test_repository.py [hunk 3]
```diff
@@ -2409,6 +2416,8 @@ def test_positive_sync_ansible_collecion_from_gallaxy(self, repo, module_org, mo
 
         :CaseImportance: High
 
+        :parametrized: yes
```
Comment (mshriver) [inline tests/foreman/cli/test_repository.py:2419]: :+1:

File: tests/foreman/cli/test_repository.py [hunk 4]
```diff
@@ -2385,17 +2385,24 @@ class TestAnsibleCollectionRepository:
...
+                'ansible-collection-auth-token': settings.ansible_hub.token,
+                'ansible-collection-auth-url': settings.ansible_hub.sso_url,
+                'ansible-collection-requirements': '{collections: \
+                                                         [redhat.satellite_operations ]}',
+            },
+        ],
+        ids=['ansible_gallaxy', 'ansible_hub'],
         indirect=True,
     )
     def test_positive_sync_ansible_collecion_from_gallaxy(self, repo, module_org, module_product):
```
Comment (mshriver) [inline tests/foreman/cli/test_repository.py:2408]: Might as well fix the typo in the test case name.

**Review by mshriver — APPROVED** [review]

File: tests/foreman/cli/test_repository.py [hunk 5]
```diff
@@ -2385,20 +2385,27 @@ class TestAnsibleCollectionRepository:
...
+                'ansible-collection-auth-url': settings.ansible_hub.sso_url,
+                'ansible-collection-requirements': '{collections: \
+                                                         [redhat.satellite_operations ]}',
+            },
+        ],
+        ids=['ansible_galaxy', 'ansible_hub'],
         indirect=True,
     )
-    def test_positive_sync_ansible_collecion_from_gallaxy(self, repo, module_org, module_product):
+    def test_positive_sync_ansible_collecion_from_galaxy(self, repo, module_org, module_product):
```
Comment (ogajduse) [inline tests/foreman/cli/test_repository.py:2408]:
```suggestion
    def test_positive_sync_ansible_collection_from_galaxy(self, repo, module_org, module_product):
```

**Review by ogajduse — APPROVED** [review]
:+1:

File: tests/foreman/cli/test_repository.py (hunk 5 above)
Comment (pondrejk) [reply tests/foreman/cli/test_repository.py:2408]: two typos? oh my :)

**Pull request merged**